Makefile

```diff
@@ -105,20 +110,35 @@ linux-shell:
 
 ### Android ###
 
+CMAKE_BUILD_DIR := cmake-build
+
 $(PREMAKE_DIR)/$(SOLUTION_NAME)_Application.mk: $(PREMAKE_DIR)/$(PREMAKE) $(wildcard $(PREMAKE_DIR)/*.lua)
 	@cd $(PREMAKE_DIR) && ./$(PREMAKE) --os=android androidmk
 	@touch $@
 
-android-configure: $(PREMAKE_DIR)/$(SOLUTION_NAME)_Application.mk
+$(CMAKE_BUILD_DIR):
+	mkdir -p $(CMAKE_BUILD_DIR)
+
+$(CMAKE_BUILD_DIR)/Makefile: $(CMAKE_BUILD_DIR) CMakeLists.txt
+	cd $(CMAKE_BUILD_DIR) && \
+	cmake \
+		-DANDROID_ABI=x86 \
+		-DCMAKE_TOOLCHAIN_FILE=${ANDROID_NDK_HOME}/build/cmake/android.toolchain.cmake \
+		-DANDROID_PLATFORM=android-21 ..
```

Comment:
The platform `android-21` seems a bit hidden in here, is this the way to go?
Would it be possible to move it up to some variable? The goal is that to be easier to spot and modify in the future.

Reply:
Added some variables, but this is just a stub now to run/test at least something.